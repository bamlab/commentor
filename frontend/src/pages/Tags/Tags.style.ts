import styled from 'styled-components';
import { colorUsage } from 'stylesheet';

export const StyledTags = styled.div`
  display: flex;
  flex-direction: column;
`;

export const LoaderContainer = styled.div`
  position: absolute;
  top: 50%;
  left: 50%;
`;

export const ErrorMessage = styled.div`
  color: ${colorUsage.error};
`;

export const FloatingButtonContainer = styled.div`
  position: fixed;
  bottom: 5px;
  width: 50em;
  left: 50%;
`;
