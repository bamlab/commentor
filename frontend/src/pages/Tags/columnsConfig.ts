import { ColumnType } from 'components/GenericTable/GenericTable.type';

export const columnsConfig: ColumnType[] = [
  {
    index: 0,
    key: 'id',
    name: 'id',
    columnWidth: 75,
  },
  {
    index: 1,
    key: 'code',
    name: 'code',
    columnWidth: 300,
  },
  {
    index: 2,
    key: 'description',
    name: 'decription',
    columnWidth: 300,
  },
  {
    index: 6,
    key: 'creationDate',
    name: 'creationDate',
    columnWidth: 200,
  },
];

export const fixedColumnCount = 0; // this is the reference
