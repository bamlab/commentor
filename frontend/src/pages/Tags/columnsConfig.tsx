import React from 'react';
import { ColumnType } from 'components/GenericTable/GenericTable.type';
import { Cell } from 'components/GenericTable/GenericTable.style';
import DeleteIcon from './components/DeleteIcon';

export const columnsConfig: ColumnType[] = [
  {
    index: 0,
    key: 'id',
    name: 'id',
    columnWidth: 75,
    renderer: (key: string, value: any, objectId: number | 'id', style?: Object): JSX.Element =>
      defaultTagCellRenderer(key, value, objectId, style),
  },
  {
    index: 1,
    key: 'code',
    name: 'code',
    columnWidth: 300,
    renderer: (key: string, value: any, objectId: number | 'id', style?: Object): JSX.Element =>
      defaultTagCellRenderer(key, value, objectId, style),
  },
  {
    index: 2,
    key: 'description',
    name: 'decription',
    columnWidth: 300,
    renderer: (key: string, value: any, objectId: number | 'id', style?: Object): JSX.Element =>
      defaultTagCellRenderer(key, value, objectId, style),
  },
  {
    index: 6,
    key: 'creationDate',
    name: 'creationDate',
    columnWidth: 200,
    renderer: (key: string, value: any, objectId: number | 'id', style?: Object): JSX.Element =>
      defaultTagCellRenderer(key, value, objectId, style),
  },
  {
    index: 7,
    key: 'delete',
    name: 'delete',
    columnWidth: 50,
    renderer: (key: string, value: any, objectId: number | 'id', style?: Object): JSX.Element =>
      IconCellRenderer(key, value, objectId, style),
  },
];

const IconCellRenderer = (
  key: string,
  value: string,
  objectId: number | 'id',
  style?: Object,
): JSX.Element => {
  return (
    <Cell key={key} style={style}>
      {objectId !== 'id' ? <DeleteIcon objectId={objectId} /> : 'Del ?'}
    </Cell>
  );
};
const defaultTagCellRenderer = (
  key: string,
  value: any,
  objectId: number | 'id',
  style?: Object,
): JSX.Element => {
  return (
    <Cell key={key} style={style}>
      {value}
    </Cell>
  );
};

export const fixedColumnCount = 0; // this is the reference
