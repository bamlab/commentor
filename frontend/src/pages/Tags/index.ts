export { default } from './Tags.wrap';
