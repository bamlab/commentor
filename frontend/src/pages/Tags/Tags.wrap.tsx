import { connect } from 'react-redux';
import { RootState } from 'redux/types';
import Tags from './Tags';
import { Dispatch } from 'redux';
import { loadTags } from 'redux/Tag/tag.actions';
import { getTags, isTagLoading } from 'redux/Tag/tag.selectors';

const mapStateToProps = (state: RootState) => ({
  tags: getTags(state),
  isTagLoading: isTagLoading(state),
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
  loadTags: () => dispatch(loadTags.request({})),
});

export default connect(
  mapStateToProps,
  mapDispatchToProps,
)(Tags);
