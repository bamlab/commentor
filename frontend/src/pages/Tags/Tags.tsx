import React, { useState, useEffect } from 'react';
import { LoaderContainer, ButtonContainer } from './Tags.style';
import Loader from 'components/Loader';
import { GenericTable } from 'components/GenericTable/GenericTable';
import { StyledTags, ErrorMessage } from './Tags.style';
import { TagType } from 'redux/Tag';
import { columnsConfig, fixedColumnCount, lineHeight, TagTableOptionsType } from './columnsConfig';
import Button from 'components/Button';
import { FormattedMessage } from 'react-intl';
import AddTagModal from './components/AddTagModal';
import UpdateTagModal from './components/UpdateTagModal';
import DeleteTagModal from './components/DeleteTagModal';

interface IProps {
  tags: TagType[];
  loadTags: () => void;
  isTagLoading: boolean;
  errorMessage: string | null;
}

const Tags = React.memo<IProps>(props => {
  const [isAddTagModalOpen, setAddTagModalValue] = useState(false);
  const [isUpdateTagModalOpen, setUpdateTagModalValue] = useState(false);
  const [isDeleteTagModalOpen, setDeleteTagModalValue] = useState(false);

  useEffect(() => {
    props.loadTags();
    // eslint-disable-next-line
  }, []);

  return (
    <StyledTags>
      <ErrorMessage>{props.errorMessage || ''}</ErrorMessage>
      <GenericTable<TagTableOptionsType>
        columnsConfig={columnsConfig}
        values={props.tags}
        fixedColumnCount={fixedColumnCount}
        defaultLineHeight={lineHeight}
        options={{
          openUpdateTagModal: () => setUpdateTagModalValue(true),
          openDeleteTagModal: () => setDeleteTagModalValue(true),
        }}
      />
      {props.isTagLoading && (
        <LoaderContainer>
          <Loader />
        </LoaderContainer>
      )}
      <ButtonContainer>
        <Button onClick={() => setAddTagModalValue(true)} disabled={props.isTagLoading}>
          <FormattedMessage id="tags.add-tag" />
        </Button>
      </ButtonContainer>
      <AddTagModal
        id="addTagModal"
        isOpen={isAddTagModalOpen}
        closeAddTagModal={() => setAddTagModalValue(false)}
      />
      <UpdateTagModal
        id="updateTagModal"
        isOpen={isUpdateTagModalOpen}
        closeUpdateModal={() => setUpdateTagModalValue(false)}
      />
      <DeleteTagModal
        id="deleteTagModal"
        isOpen={isDeleteTagModalOpen}
        closeRemoveTageModal={() => setDeleteTagModalValue(false)}
      />
    </StyledTags>
  );
});

export default Tags;
