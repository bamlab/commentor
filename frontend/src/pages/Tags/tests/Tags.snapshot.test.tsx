import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Tags from '../Tags';

/*
Snapshot tests allow you to easily lock the comportment of a component.
Given props, it renders the component and compares it to the saved snapshot.
*/
describe('[Snapshot] <Tags />', () => {
  const props = {
    tags: [],
    loadTags: jest.fn(),
    isTagLoading: true,
    addTag: jest.fn(),
  };
  it('should render Tags page', () => {
    const wrapper = shallow(<Tags {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
