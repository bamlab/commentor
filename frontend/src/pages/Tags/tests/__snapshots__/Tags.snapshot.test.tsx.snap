// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Snapshot] <Tags /> should render a button with a label 1`] = `
<Tagsstyle__StyledTags>
  <Button>
    <FormattedMessage
      id="tags.add-tag"
      values={Object {}}
    />
  </Button>
  <GenericTable
    columnsConfig={
      Array [
        Object {
          "columnWidth": 75,
          "index": 0,
          "key": "id",
          "name": "id",
        },
        Object {
          "columnWidth": 300,
          "index": 1,
          "key": "code",
          "name": "code",
        },
        Object {
          "columnWidth": 300,
          "index": 2,
          "key": "description",
          "name": "decription",
        },
        Object {
          "columnWidth": 200,
          "index": 6,
          "key": "creationDate",
          "name": "creationDate",
        },
      ]
    }
    fixedColumnCount={2}
  />
</Tagsstyle__StyledTags>
`;
